tests/integration/Core/Repository/Values/User/Limitation/ChangeOwnerLimitationTest.php

```diff
@@ -0,0 +1,192 @@
+    public function deniedProvider(): array
+    {
+        $otherUserId1 = 123;
+        $otherUserId2 = 456;
+        $otherUserId3 = 789;
+
+        return [
+            [$otherUserId1, [-1]],
+            [$otherUserId1, [$otherUserId2, $otherUserId3]],
+        ];
+    }
+
+    /**
+     * @dataProvider allowedProvider
+     *
+     * @param int[] $limitationValues
+     */
+    public function testChangeOwnerLimitationAllowed(?int $ownerId, array $limitationValues): void
+    {
+        /* BEGIN: Use Case */
+        $currentUser = $this->createUserVersion1('current_user', null, null, 42);
+        $ownerId = $ownerId ?? $currentUser->id;
+
+        $draft = $this->createTestCaseDraft($currentUser, $ownerId, $limitationValues);
+        /* END: Use Case */
+
+        self::assertInstanceOf(Content::class, $draft);
+    }
+
+    /**
+     * @dataProvider deniedProvider
+     *
+     * @param int[] $limitationValues
+     */
+    public function testChangeOwnerLimitationDenied(?int $ownerId, array $limitationValues): void
+    {
+        self::expectException(UnauthorizedException::class);
+
+        /* BEGIN: Use Case */
+        $currentUser = $this->createUserVersion1('current_user', null, null, 42);
+        $ownerId = $ownerId ?? $currentUser->id;
+
+        $this->createTestCaseDraft($currentUser, $ownerId, $limitationValues);
+        /* END: Use Case */
+    }
+
+    /**
+     * @param int[] $limitationValues
+     */
+    private function createTestCaseDraft(
+        User $currentUser,
+        int $ownerId,
+        array $limitationValues
+    ): Content {
+        $role = $this->updateRoleWithChangeOwnerLimitation(
+            $this->roleService->loadRoleByIdentifier('Anonymous'),
+            $limitationValues
+        );
+        $this->roleService->assignRoleToUser($role, $currentUser);
+        $this->permissionResolver->setCurrentUserReference($currentUser);
+
+        $contentCreateStruct = $this->createWikiPageContentCreateStruct($ownerId);
+        $locationCreateStruct = $this->createWikiPageLocationCreateStruct($this->generateId('location', 2));
+
+        return $this->contentService->createContent($contentCreateStruct, [$locationCreateStruct]);
+    }
+
+    /**
+     * @throws \ErrorException
+     */
+    private function getContentCreatePolicyDraft(RoleDraft $role): PolicyDraft
+    {
+        foreach ($role->getPolicies() as $policy) {
+            if ($policy->module === 'content' && $policy->function === 'create') {
+                return $policy;
+            }
+        }
+
+        throw new ErrorException('No content:create policy found.');
+    }
+
+    private function hasContentCreatePolicy(RoleDraft $role): bool
+    {
+        foreach ($role->getPolicies() as $policy) {
+            if ($policy->module === 'content' && $policy->function === 'create') {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param int[] $limitationValues
+     */
```

Comment:
Nice to have
```suggestion
     *
     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\Exception
     */
```
assuming you apply previous comment, otherwise need to add also `@throws \ErrorException`